kong/db/declarative/migrations/route_path.lua

```diff
@@ -2,31 +2,56 @@ local migrate_path = require "kong.db.migrations.migrate_path_280_300"
 
 local pairs  = pairs
 local ipairs = ipairs
-local null   = ngx.null
 
-return function(tbl, version)
-  if not tbl or not (version == "1.1" or version == "2.1") then
-    return
+local empty_tbl = {}
+
+local function indexable(val)
+  if type(val) == "table" then
+    return true
   end
 
-  local routes = tbl.routes
+  local meta = getmetatable(val)
+  return meta and meta.__index
+end
 
-  if not routes then
-    -- no need to migrate
-    return
+local function table_default(val)
+  -- we cannot verify it with type(val) == "table" because
+  -- we may recieve indexable cdata/lightuserdata
```

Comment:
>  -- we may recieve indexable cdata/lightuserdata

When and why would this happen?

Reply:
For the same reason, I'm using pairs instead of pairs. But let's give it a try and revert if it fails.